File: Programmers/JiHo/11주차/모음사전.py
def solution(word):
    li = {'A':0, 'E':1, 'I':2, 'O':3, 'U':4}
    e = [781,156,31,6,1]
    answer = 0
    count = 0
    for i in word :
        answer += li[i] * e[count]
        answer += 1
        count += 1

    return answer

File: Programmers/BoHwa/11주차/모음사전.java
class Solution {
    static char[] aeiou = {'A','E','I','O','U'};
    static ArrayList<String> list;
    
    public int solution(String word) {
        int answer = 0;
        list=new ArrayList<>();
        
        stick(0,"");
        Collections.sort(list);
        answer=list.indexOf(word)+1;
        
        return answer;
    }
    public void stick(int n, String st){
        if (n>=5) return;
        for (int i=0; i<aeiou.length; i++){
            list.add(st+aeiou[i]);
            stick(n+1,st+aeiou[i]);
        }
        
    }
}
